src/Adapter/Product/CommandHandler/BulkDeleteProductHandler.php

```diff
@@ -63,7 +64,12 @@ public function handle(BulkDeleteProductCommand $command): void
 
     protected function handleSingleAction(ProductId $productId, $command = null)
     {
-        $this->productRepository->delete($productId);
+        //@todo: not sure about all shops constraint. Something might be confused here.
+        // Im pretty sure it used to delete product only in context shop when we don't provide shops explicitly.
+        // but in ProductController I see it used in bulkDeleteFromAllShopsAction.
+        // So this whole command might deserve renaming, or its confused with DeleteProductFromShopsCommand (which seems to be unused at all)
+        // need to double check this part before merging
+        $this->productRepository->deleteByShopConstraint($productId, ShopConstraint::allShops());
```

Comment:
ping @jolelievre same question related to your recently introduced DeleteProductFromShops command.